assignment2/assignment2.py

```diff
@@ -0,0 +1,143 @@
+"""String Compression: Implement a method to perform basic string compression using the counts
+of repeated characters. For example, the string a a b c c c c c a a a would become a 2 b l c 5 a 3 , If the
+"compressed" string would not become smaller than the original string, your method should return
+the original string. You can assume the string has only uppercase and lowercase letters (a - z)."""
+
+def compress(string: str) -> str:
+    res = ""
+    letters = list()
+    nums = list()
+    letter = 0
+
+    for i in range(len(string)):
+        letter += 1
+        if i+1 < len(string) and string[i] == string[i+1]: continue
+        else:
+            letters.append(string[i])
+            nums.append(str(letter))
+            letter = 0
+
+    for i in range(len(letters)):
+        res += letters[i]
+        res += nums[i]
+    
+    return res
+
+print("-----------------------------------")
+print("aabcccccaaa")
+print(compress("aabcccccaaa"))
+print("-----------------------------------")
+
+"""Delete Middle Node: Implement an algorithm to delete a node in the middle (i.e., any node but
+the first and last node, not necessarily the exact middle) of a singly linked list, given only access to
+that node."""
+
+class Node:
+    def __init__(self, next, data):
+        self.next = next
+        self.data = data
+    
+    def append(self, data):
+        child = Node(None, data)
+        current = self
+        while current.next != None:
```

Comment:
Because `None` is already falsey in Python, you don't need to check explicitly against it here.
```suggestion
        while current.next:
```